Sources/TuistGenerator/Utils/NamespaceGenerator.swift

```diff
@@ -0,0 +1,84 @@
+import PathKit
+import StencilSwiftKit
+import SwiftGenKit
+import TSCBasic
+import TuistSupport
+
+enum NamespaceType {
+    case assets
+    case strings
+
+    var templateFileName: String {
+        switch self {
+        case .assets:
+            return "xcassets.stencil"
+        case .strings:
+            return "strings.stencil"
+        }
+    }
+}
+
+private extension NamespaceType {
+    func parser() throws -> Parser {
+        switch self {
+        case .assets:
+            return try AssetsCatalog.Parser()
+        case .strings:
+            return try Strings.Parser()
+        }
+    }
+}
+
+protocol NamespaceGenerating {
```

Comment:
Nitpick: `SynthesizedResouceInterfacesGenerating`.

Reply:
Yep, I'll rename it throughout the PR!